src/amo/sagas/addonsByAuthors.js

```diff
@@ -0,0 +1,60 @@
+// Disabled because of
+// https://github.com/benmosher/eslint-plugin-import/issues/793
+/* eslint-disable import/order */
+import { all, call, put, select, takeEvery } from 'redux-saga/effects';
+/* eslint-enable import/order */
+
+import {
+  FETCH_OTHER_ADDONS_BY_AUTHORS,
+  loadOtherAddonsByAuthors,
+} from 'amo/reducers/addonsByAuthors';
+import searchApi from 'core/api/search';
+import log from 'core/logger';
+import { createErrorHandler, getState } from 'core/sagas/utils';
+
+
+const NB_ADDONS_MAX_PER_AUTHOR = 6;
+
+export function* fetchOtherAddonsByAuthors({ payload }) {
+  const { errorHandlerId, authors, slug } = payload;
+  const errorHandler = createErrorHandler(errorHandlerId);
+
+  yield put(errorHandler.createClearingAction());
+
+  try {
+    const state = yield select(getState);
+
+    const responses = yield all(authors.map((author) => call(searchApi, {
+      api: state.api,
+      filters: { author, page_size: NB_ADDONS_MAX_PER_AUTHOR },
+    })));
+
+    // This computes a map of add-ons based on the N previous API calls (N
+    // being the number of authors). It eliminates duplicates (_.e.g._ 3 people
+    // authored the same add-on results in 3 times the same add-on). It also
+    // omits the add-on (identified by `slug`) that is related to these other
+    // add-ons, so that we will not list the same add-on already displayed in
+    // the detail view.
+    const addons = responses.reduce((previousAddons, response) => {
```

Comment:
I don't think this is worth the effort and complexity (and added test coverage). The reducer will actually do this naturally because it is creating a map of add-ons by ID (slug). Also, if we get an API patch for filtering by multiple authors then we won't need it at all.

Reply:
If we dont get the filter for a list of authors, then I am afraid we are forced to do that. We need a distinct set of add-ons and I think the saga should handle this.

Reply:
Oh, wait, I didn't read the reducer code close enough. I thought it was storing a *map* of add-ons but I see it's storing an array of them. In any case, I find that this sort of code should live in the reducer because that's its job anyway.